source/slang/slang-compiler.cpp

```diff
@@ -1103,13 +1141,15 @@ SlangResult dissassembleDXILUsingDXC(
         Int                     entryPointIndex,
         TargetRequest*          targetReq,
         EndToEndCompileRequest* endToEndReq,
-        List<uint8_t>&          binOut)
+        ComPtr<ISlangSharedLibrary>& outSharedLib,
+        List<uint8_t>&          outBin)
```

Comment:
This is another bit of a oddity in the design: this one function now produces two different outputs, depending on the target.

It might be nice if this was factored into a core worker routine that produces the "raw" output (which I assume is the shared library file and the rest of the "file set" associated with it), and then two wrapper routines that take that "raw" output and produce what the  two main cases need (in one case a loaded `ISlangSharedLibrary` and in the other case a binary blob).

Reply:
This is a little clunky. That I didn't err on the side of returning a say IUnknown - such there was one 'thing' that could have many faces because that seemed to be more confusing that this such that it us. (Ie it would be ISharedLibrary or IBlob depending).

It may not be too easy to break up the functionality - unless some kind of 'proxy' of the state can be emitted. Because part of the aim here was once the binary file is produced to just be able to use it. You could avoid all that by always creating in memory (by loading) and then resaving etc - but in that case the cure seems way worse than the disease.

Reply:
> You could avoid all that by always creating in memory (by loading) and then resaving etc - but in that case the cure seems way worse than the disease.

I kind of assumed it would work the other way around: the code would default to producing a file (set), since that is what happens when we invoke the downstream compiler, and then the path that *wants* the in-memory blob would load it and delete the files, while the other path would just use those files as-is.

Reply:
Ok I see - well that's an easier to implement possibility, although then has the assumption that that's how C/C++ backend compilers always work...

Reply:
I figured that when that *isn't* the case we'll just refactor the code yet again. :)